app/controllers/HasCorrectGoodsController.scala

```diff
@@ -140,13 +154,115 @@ class HasCorrectGoodsController @Inject() (
+                                      goodsDescription = oldRecord.goodsDescription,
+                                      countryOfOrigin = oldRecord.countryOfOrigin,
+                                      category = None,
+                                      assessments = oldRecord.assessments,
+                                      supplementaryUnit = oldRecord.supplementaryUnit,
+                                      measurementUnit = oldRecord.measurementUnit,
+                                      comcodeEffectiveFromDate = oldRecord.comcodeEffectiveFromDate,
+                                      comcodeEffectiveToDate = oldRecord.comcodeEffectiveToDate
+                                    )
+                                  )
+      _                         = auditService.auditFinishUpdateGoodsRecord(recordId, request.affinityGroup, updateGoodsRecord)
+      _                        <- updateGoodsRecordIfPutValueChanged(
+                                    commodity.commodityCode,
+                                    oldRecord.comcode,
+                                    updateGoodsRecord,
+                                    putGoodsRecord
+                                  )
+      updatedAnswersWithChange <-
+        Future.fromTry(request.userAnswers.remove(HasCommodityCodeChangePage(recordId)))
+      updatedAnswers           <- Future.fromTry(updatedAnswersWithChange.remove(CommodityCodeUpdatePage(recordId)))
+      _                        <- sessionRepository.set(updatedAnswers)
+    } yield Redirect(
+      navigator.nextPage(HasCorrectGoodsCommodityCodeUpdatePage(recordId), NormalMode, updatedAnswers)
+    )).recover { case e: GoodsRecordBuildFailure =>
+      logErrorsAndContinue(
+        e.getMessage,
+        controllers.goodsRecord.routes.SingleRecordController.onPageLoad(recordId)
+      )
+    }
+
+  private def handleValidateError[T](result: EitherNec[ValidationError, T]): Future[T] =
+    result match {
+      case Right(value) => Future.successful(value)
+      case Left(errors) => Future.failed(GoodsRecordBuildFailure(errors))
+    }
+
+  private case class GoodsRecordBuildFailure(errors: data.NonEmptyChain[ValidationError]) extends Exception {
+    private val errorsAsString       = errors.toChain.toList.map(_.message).mkString(", ")
+    private val errorMessage: String = "Unable to update Goods Record."
+
+    override def getMessage: String = s"$errorMessage Missing pages: $errorsAsString"
+  }
+
+  private def updateGoodsRecordIfPutValueChanged(
+    newValue: String,
+    oldValue: String,
+    newUpdateGoodsRecord: UpdateGoodsRecord,
+    putRecordRequest: PutRecordRequest
+  )(implicit hc: HeaderCarrier): Future[Done] =
+    if (newValue != oldValue) {
+
+      if (config.useEisPatchMethod) {
+        goodsRecordConnector.putGoodsRecord(
+          putRecordRequest,
+          newUpdateGoodsRecord.recordId
+        )
+      } else {
+        goodsRecordConnector.updateGoodsRecord(
```

Comment:
it looks like there is a missing test here / not covered

Reply:
@JemmaEdwards Tests added